src/main/java/com/esri/samples/raster/raster_rendering_rule/RasterRenderingRuleSample.java

```diff
@@ -0,0 +1,166 @@
+
+import java.util.List;
+
+import com.esri.arcgisruntime.arcgisservices.RenderingRuleInfo;
+import com.esri.arcgisruntime.layers.RasterLayer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.raster.ImageServiceRaster;
+import com.esri.arcgisruntime.raster.RenderingRule;
+
+import javafx.application.Application;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Scene;
+import javafx.scene.control.ComboBox;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
+
+public class RasterRenderingRuleSample extends Application {
+
+    private ComboBox<String> renderingRuleDropDownMenu;
+    private MapView mapView;
+    private ArcGISMap map;
+    private final String imageServiceUrl = "http://sampleserver6.arcgisonline.com/arcgis/rest/services/CharlotteLAS/ImageServer";
+
+    @Override
+    public void start(Stage stage) {
+        try {
+            // create stack pane and application scene
+            StackPane stackPane = new StackPane();
+            Scene scene = new Scene(stackPane);
+
+            // set title, size, and add scene to stage
+            stage.setTitle("Raster Rendering Rule Sample");
+            stage.setWidth(800);
+            stage.setHeight(700);
+            stage.setScene(scene);
+            stage.show();
+
+            // create drop down menu of Rendering Rules
+            renderingRuleDropDownMenu = new ComboBox<>();
+            renderingRuleDropDownMenu.setPromptText("Select a Raster Rendering Rule");
+            renderingRuleDropDownMenu.setEditable(false);
+            renderingRuleDropDownMenu.setMaxWidth(260.0);
+
+            // create a Streets BaseMap
+            map = new ArcGISMap(Basemap.createStreets());
+
+            // add the map to a new map view
+            mapView = new MapView();
+            mapView.setMap(map);
+
+            // create an Image Service Raster as a raster layer and add to map
+            final ImageServiceRaster imageServiceRaster = new ImageServiceRaster(imageServiceUrl);
+            final RasterLayer imageRasterLayer = new RasterLayer(imageServiceRaster);
+            map.getOperationalLayers().add(imageRasterLayer);
+
+            imageServiceRaster.loadAsync();
+
+            // add event listener to loading of Image Service Raster and wait until loaded
+            imageRasterLayer.addDoneLoadingListener(() -> {
+                if (imageRasterLayer.getLoadStatus() == LoadStatus.LOADED){
+                    // zoom to extent of the raster
+                    mapView.setViewpointGeometryAsync(imageServiceRaster.getServiceInfo().getFullExtent());
+
+                    // get the predefined rendering rules
+                    List<RenderingRuleInfo> renderingRuleInfos = imageServiceRaster.getServiceInfo().getRenderingRuleInfos();
```

Comment:
I noticed the rendering rule has a description as well which is pretty useful. Maybe that could be displayed in a label somewhere? Would help users know what the rendering rule is doing.

Reply:
Good call! I've built a `VBox` for the `ComboBox` and `Label` for the description to live in. Would be nice to have the `VBox` resize dynamically to fit the text. Any tips on how that could be done?